statistics/scalar.go

```diff
@@ -288,3 +288,33 @@ func enumRangeValues(low, high types.Datum, lowExclude, highExclude bool) []type
 	}
 	return nil
 }
+
+// calcWidth is used to calculate the width of the interval [lower, upper] that lies within the [lower, value]
+// using the continuous-value assumption.
+func calcWidth(lower, upper float64) float64 {
+	if upper <= lower {
+		return 0.0
+	}
+	return upper - lower
+}
+
+func (hg *Histogram) calcStatWidth(index int, value *types.Datum) float64 {
+	lower, upper := hg.Bounds.GetRow(2*index), hg.Bounds.GetRow(2*index+1)
```

Comment:
The index is unnecessary here and we should get the lower and upper by this way:
```
lower := hg.Bounds.GetRow(0) // the first bucket's left bound
upper := hg.Bounds.GetRow(2*(len(hg.Buckets)-1)+1) // the last bucket's right bound
```